src/models/user.ts

```diff
@@ -0,0 +1,181 @@
+import { query, client } from '../db/database';
+import Role from './role';
+import UserToken from './userToken';
+import { StatusCodeError } from '../handlers/error-handler';
+
+class User {
+  constructor(
+    private username: string,
+    private password: string,
+    private id: number | null = null,
+    private created_at: Date | null = null,
+    private deleted_at: Date | null = null,
+    private etag: string = '',
+    private roles: string[] = []
+  ) {}
+
+  async save() {
+    const operation = this.id !== null ? 'update' : 'save';
+
+    const transactionClient = await client();
+
+    try {
+      await transactionClient.query('BEGIN');
+
+      switch (operation) {
```

Comment:
Please refactor it and remove switch. Instead of it separate save and update actions into external functions.